modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/AddGadgetToDashboardTest.java

```diff
@@ -62,22 +62,61 @@ public void tearDown() throws Exception {
         }
     }
 
+    @Test(groups = "wso2.ds.dashboard", description = "Adding blocks to an existing dashboard")
+    public void testAddBlocks() throws Exception {
 
-    @Test(groups = "wso2.ds.dashboard", description = "Adding gadgets to an existing dashboard from dashboard server")
-    public void testAddGadgetToDashboard() throws Exception {
         DSWebDriver driver = getDriver();
-        String[][] gadgetMappings = {{"g1", "b"}, {"usa-map", "c"}};
-        String script = generateAddGadgetScript(gadgetMappings);
 
         driver.get(getBaseUrl() + "/portal/dashboards");
+
         driver.findElement(By.cssSelector("#" + DASHBOARD_TITLE + " a.ues-edit")).click();
+        driver.findElement(By.cssSelector("#ues-add-block-menu-item > a")).click();
+        driver.findElement(By.id("ues-add-block-btn")).click();
+
+        driver.findElement(By.cssSelector("a.ues-dashboard-preview")).click();
+        pushWindow();
+
+        assertTrue(isBlockPresent("a"), "The block 'a' does not exist");
+        driver.close();
+        popWindow();
+    }
+
+    @Test(groups = "wso2.ds.dashboard", description = "Removing blocks from an existing dashboard",
+            dependsOnMethods = "testAddBlocks")
+    public void testRemoveBlock() throws MalformedURLException, XPathExpressionException, InterruptedException {
+
+        DSWebDriver driver = getDriver();
+        driver.findElement(By.cssSelector("#a.ues-component-box .ues-component-box-remove-handle")).click();
+
+        // TODO: change the behaviour in the dashboard to reflect the change after saving the change. Then remove sleep
+        Thread.sleep(500);
```

Comment:
Get rid of Thread.sleep() by driver.findElement which wraps Webdriver.wait and you can set "maximumWaitingTime" time in automation.xml